packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -263,14 +269,25 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 
 		// Anonymize the commits from this transaction by stripping their revision tags.
 		// Otherwise, the change rebaser will record their tags and those tags no longer exist.
-		const anonymousCommits = commits.map(({ change }) => ({ change, revision: undefined }));
+		const anonymousCommits = commits.map((commit) => ({
+			change: this.changeFamily.rebaser.replaceRevisions(
+				commit.change,
+				new Set([commit.revision]),
+				undefined,
+			),
+			revision: undefined,
+		}));
 		// Squash the changes and make the squash commit the new head of this branch
 		const squashedChange = this.changeFamily.rebaser.compose(anonymousCommits);
 		const revision = this.mintRevisionTag();
 
 		const newHead = mintCommit(startCommit, {
 			revision,
-			change: squashedChange,
+			change: this.changeFamily.rebaser.replaceRevisions(
+				squashedChange,
+				new Set([undefined]),
+				revision,
+			),
```

Comment:
I'm curious what motivated this two-step approach as opposed to either calling replaceRevisions on each commit with the final newRevision, or calling replaceRevisions on the composed commit with the set of all input commits as the oldRevision.